android/app/src/main/java/org/uwaterloo/subletr/pages/watcardverification/WatcardVerificationPageView.kt

```diff
@@ -0,0 +1,254 @@
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rxjava3.subscribeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.core.app.ActivityCompat.startActivityForResult
+import androidx.core.content.ContextCompat.startActivity
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.theme.SubletrTheme
+import org.uwaterloo.subletr.theme.darkerGrayButtonColor
+import org.uwaterloo.subletr.theme.lightGrayButtonColor
+import org.uwaterloo.subletr.theme.secondaryTextColor
+import org.uwaterloo.subletr.theme.subletrPink
+
+private const val ELEMENT_WIDTH = 0.75f
+@Composable
+fun WatcardVerificationPageView(
+	modifier : Modifier = Modifier,
+	viewModel : WatcardVerificationPageViewModel = hiltViewModel(),
+	uiState: WatcardVerificationUiState = viewModel.uiStateStream.subscribeAsState(
+		WatcardVerificationUiState.Loading
+	).value
+) {
+
+	if (uiState is WatcardVerificationUiState.Loading) {
+		Column(
+			modifier = modifier
+				.fillMaxSize(1.0f)
+				.imePadding(),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			CircularProgressIndicator()
+		}
+	} else if (uiState is WatcardVerificationUiState.Loaded) {
+		Column(
+			modifier = modifier.fillMaxSize(1.0f),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			Spacer(modifier = modifier.weight(8.0f))
+			
+			Text(
+				text = stringResource(id = R.string.watcard_verification),
+				style = MaterialTheme.typography.titleMedium,
+			)
+
+			Column(modifier = modifier
+				.fillMaxWidth(1.0f)
+				.padding(horizontal = dimensionResource(id = R.dimen.l))
+			) {
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt1),
+					color = secondaryTextColor,
+				)
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt2),
+					color = secondaryTextColor,
+				)
+			}
+
+			setContent(
+				viewModel = viewModel,
+				watcard = uiState.watcard
+			)
+
+			SetUploadButton(
+				viewModel,
+				uiState.watcard
+			)
+
+			Spacer(modifier = modifier.weight(0.5f))
```

Comment:
the spacing between the buttons isn't enough here, maybe have a set spacing using
`Spacer(modifier = Modifier.height(dimensionResource(id = R.dimen.s)))`